Automattic/wp-calypso
# wp-admin Site Default: Add "Default to wp-admin" toggle to Hosting Configuration page

Related to https://github.com/Automattic/dotcom-forge/issues/4154

## Proposed Changes

This PR does the following:

* adds a new `site-wp-admin-card` with a toggle
* when the toggle is toggled on, it sends an API request to set Atomic Persistent data and sets the site option to `wp-admin`
* when the toggle is toggled off, it sends an API request to set Atomic Persistent data and sets the site option to `calypso`
* there is a text on the frontend that outputs the option that is set when change the toggle (it will likely be removed in the future; at this point, it is a demonstration that the toggle is working as expected)
* it hides the card behind the feature flag

<img width="1259" alt="Screenshot 2023-10-18 at 5 04 54 PM" src="https://github.com/Automattic/wp-calypso/assets/25575134/53c33af0-585b-41c1-a927-9f6e6dc23a44">

## Testing Instructions

* Make sure you have the most recent trunk before starting (https://github.com/Automattic/jetpack/pull/33610/files)
* Apply this diff to your sandbox (if not deployed): D125015-code
* Sandbox the public API
* Pull changes from this branch (or use Calypso Live)
* Ensure that you have an Atomic site
* Navigate to `https://wordpress.com/hosting-config/yourAtomicSite`
* Scroll to the bottom of the page and observe the `wp-admin` card
* Initially, the text `Currently, the activated mode is` will be empty - this is expected (the text will be removed later one, at the moment it is just returning the value of the option`
* Toggle the setting on 
* Confirm that the text changes to `Currently, the activated mode is wp-admin` (it might take a second to appear)
* Check the Network tab and confirm that the request is sent as expected with `wp-admin`:

<img width="1266" alt="Screenshot 2023-10-18 at 5 09 53 PM" src="https://github.com/Automattic/wp-calypso/assets/25575134/20ac7169-d50a-4879-b857-3e2cfed354f1">

* Refresh the page and confirm that the setting stays on the toggle and the toggle does not get turned off
* Toggle the setting off
* Confirm that the text changes to `Currently, the activated mode is calypso`
* Check the Network tab and confirm that the request is sent as expected
* Refresh the page and confirm the setting stays

I think this generally covers the happy path. Some things that will need to be added in the future iteration:

* Test coverage
* Some error handling (currently, I am reverting to a previous setting without displaying an error message)
* Possibly a loading state if we think we need it
* Design improvements

## Pre-merge Checklist
